cli/args/package_json.rs

```diff
@@ -26,7 +25,7 @@ pub enum PackageJsonDepValueParseError {
 }
 
 pub type PackageJsonDeps =
-  BTreeMap<String, Result<PackageReq, PackageJsonDepValueParseError>>;
+  IndexMap<String, Result<PackageReq, PackageJsonDepValueParseError>>;
```

Comment:
Changed this to an `IndexMap` so that dependencies are favoured over dev dependencies when resolving an npm specifier to a package.json dep or dev dep with byonm.